### ISSUE TITLE
Add helpers to get line/col of error message

### ISSUE THREAD

**Issue by lihaoyi** [lihaoyi/fastparse#46]
`index` is enough for machines but hard to read for humans. It's trivial to get the line/col via

```scala
val lines = f.input.take(f.index).lines.toVector
val line = lines.length 
val col = lines.last.length
```

But it's probably worth chucking this (or some more-optimized version of it) into the `Failure` object for everybody's convenience

**Comment by Alain-Bearez** [comment]
Would you want to also report the parser position as ```line:column``` in ```Failure.formatParser```? Most calls to this function are not enclosed in the scope of a ```Failure```.

It might be the case to push some additional information into the ```ParseCtx```, like the number of already read lines and the global input index at start of current line.

**Comment by lihaoyi** [comment]
reporting `line:column` in `formatParser` seems reasonable. I'm not sure if we still want to show the `index` somewhere; it seems like it might be useful but maybe it isn't

You shouldn't need too much additional info; IIRC most places where we call `formatParser` already have access to the `input: String`, and that + `index` is enough to re-derive the line and column

**Comment by lihaoyi** [comment]
This has landed in #52. Thanks @Alain-Bearez!
